src/java/org/apache/cassandra/db/compaction/RepairFinishedCompactionTask.java

```diff
@@ -90,7 +89,8 @@ protected void runMayThrow() throws Exception
         {
             if (obsoleteSSTables)
             {
-                transaction.finish();
+                transaction.prepareToCommit();
+                transaction.commit();
```

Comment:
What is the purpose of making this change? It looks like `finish()` calls these methods by default but could be overridden, so after a quick glance, it's not clear that this is a safe replacement.